### ISSUE TITLE
Parser fails with binary message / null bytes

### ISSUE THREAD

**Issue by felixbuenemann** [hit9/spp_py#1]
I have a SSDB database where both keys and values are binary and can contain null bytes (key is a packed 64-bit BE int timestamp, followed by a binary UUIDv4, value is MessagePack-serialized data).

Python 2.7 Example:

``` python
from spp import Parser

# reply to ssdb.keys('', '', 10)
buf=b'2\nok\n24\n\x00\x05!=\x95\'\xe2\xb6kH*\x9c\xce\x8cI\xab\xb03C)]u\n\x80\n24\n\x00\x05!=\x95\'\xe8\x14N\x12W\x9a\xaa\xa3Dv\x8ba\xc1(:\xb9O\x95\n24\n\x00\x05!=\x95\'\xeaH\xe0C\x1c\x1e\x8f\xbbI\xaf\xac\xa9\xf7\xf1\xfe\x15\xb9\xda\n24\n\x00\x05!=\x95\'\xf1F\x81"\xd8\xf4\x18?I6\x8d\xd5\xfbh/2\xaa\xb7\n24\n\x00\x05!=\x95\'\xf9\x91+\x15<\xa1\xcd\x1eD\xc6\x8a(\xb8x]\x91;\xa8\n24\n\x00\x05!=\x95\'\xfb\x94\x964\xc25\xb4\rJ\x18\x9b\x83=\x98\x90l\x9f\xda\n24\n\x00\x05!=\x95(\x02f\xa5P\xdb\xf4\x8bDN\xbc\xaa\x06\xe0\x1b\xfdeM\xc9\n24\n\x00\x05!=\xd0l\x8c\xf8\x834Di\x81\x93O\xf0\xac\xc5\xc1J$!\xb6J\n24\n\x00\x05!=\xd1V\xa6M<s\x98\xcb\x8b\xe0B\xf5\x8d\t\xb2\x18;!\xac>\n24\n\x00\x05!=\xe4U\x9d\xf2\xdb\xeb,\xba\x80PJh\x92\xf2\xf2\x89\xd0(\xa8\x9e\n\n'

parser = Parser()
parser.feed(str(buf))
# returns 8 (or other small int)

parser.get()
# returns None
```

Because of this problem the request() while loop in ssdb.py hangs on a blocking read, so it appears that commands randomly hang.

The buffer should contain 10 binary keys, each 24 bytes (8bytes uint64be posix timestamp, 16bytes uuid).

Versions used:
Mac OS X 10.11.1
ssdb-server 1.9.2
Python 2.7.10
spp 0.0.6

**Comment by felixbuenemann** [comment]
If I replace all null bytes with `\x01`, I can parse the message successfully.

~~I've looked through the sources, and the `hbuf_str` method seems suspicious:
https://github.com/hit9/spp_py/blob/master/src/c/hbuf.c#L103~~

~~This method returns early if is finds a null byte.~~

**Comment by felixbuenemann** [comment]
I've looked at the code in more detail and I can see that `Parser.feed` calls `spp_feed`, which calls `hbuf_puts`, where `strlen` is called. Because `strlen` works with null terminated strings this cut's of the message. I think the right approach would be to pass the size of the data from spp_feed through hbuf_puts to hbuf_put as a parameter.

**Comment by felixbuenemann** [comment]
@hit9 I've fixed the issue described in PR #2 and tested with ssdb.py, please review.

**Comment by felixbuenemann** [comment]
Fixed in spp_py 0.0.7.
